lib/TH/generic/THStorage.h

```diff
@@ -28,6 +29,8 @@ typedef struct THStorage
     char flag;
     THAllocator *allocator;
     void *allocatorContext;
+    int type;
```

Comment:
The type of an element is already encoded in the type of the C struct: `THStorage` is part of a macro that gets expanded to `THFloatStorage`, `THByteStorage`, etc. See [here](https://github.com/torch/torch7/blob/4fd73c14285dc4a4d787276dd78dd1ef63542628/lib/TH/THStorage.h#L7).